compiler/src/iree/compiler/InputConversion/MHLO/MHLOToLinalgOnTensors.cpp

```diff
@@ -228,8 +228,14 @@ struct OptimizationBarrierOpConversion
   LogicalResult matchAndRewrite(
       mhlo::OptimizationBarrierOp op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<IREE::Util::OptimizationBarrierOp>(
-        op, op.getOperands());
+    SmallVector<Value> outputs;
+    for (unsigned i = 0, e = adaptor.getOperands().size(); i < e; ++i) {
```

Comment:
I think we can iterate on `adpator.getOperands()`, which is simpler.

```c++
for (auto operand : adaptor.getOperands()) {
  // .... create ... operand ...
}
```

Reply:
Good point, thanks! Fixed.